CMakeLists.txt

```diff
@@ -786,41 +789,41 @@ if (MPI)
+    install(FILES ICB/parpack.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
+    install(FILES ICB/parpack.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
   endif()
   if (ICBEXMM)
-    install(FILES EXAMPLES/MATRIX_MARKET/arpackSolver.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
+    install(FILES EXAMPLES/MATRIX_MARKET/arpackSolver.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
     install(FILES "${PROJECT_BINARY_DIR}/EXAMPLES/MATRIX_MARKET/arpackSolver.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
   endif()
 endif()
 
-install(FILES debug.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
+install(FILES debug.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
 if(ICB)
-  install(FILES ICB/debug_c.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
-  install(FILES ICB/debug_c.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
+  install(FILES ICB/debug_c.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
+  install(FILES ICB/debug_c.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
 endif()
 
-install(FILES stat.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
+install(FILES stat.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
 if(ICB)
-  install(FILES ICB/stat_c.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
-  install(FILES ICB/stat_c.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
+  install(FILES ICB/stat_c.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
+  install(FILES ICB/stat_c.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
 endif()
 
 configure_file(arpackdef.h.in "${PROJECT_BINARY_DIR}/arpackdef.h" @ONLY)
-install(FILES "${PROJECT_BINARY_DIR}/arpackdef.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
+install(FILES "${PROJECT_BINARY_DIR}/arpackdef.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
 
 configure_file(arpackicb.h.in "${PROJECT_BINARY_DIR}/arpackicb.h" @ONLY)
-install(FILES "${PROJECT_BINARY_DIR}/arpackicb.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack)
+install(FILES "${PROJECT_BINARY_DIR}/arpackicb.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/arpack${SUFFIX64})
 
 # Provide find_package for arpack-ng to users.
 configure_file(arpack-ng-config.cmake.in "${PROJECT_BINARY_DIR}/arpack-ng-config.cmake" @ONLY)
-install(FILES "${PROJECT_BINARY_DIR}/arpack-ng-config.cmake" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/arpack-ng) # find_package(arpack-ng)
+install(FILES "${PROJECT_BINARY_DIR}/arpack-ng-config.cmake" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/arpack${SUFFIX64}-ng) # find_package(arpack-ng)
 configure_file(arpack-ng-config-version.cmake.in "${PROJECT_BINARY_DIR}/arpack-ng-config-version.cmake" @ONLY)
-install(FILES "${PROJECT_BINARY_DIR}/arpack-ng-config-version.cmake" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/arpack-ng)
+install(FILES "${PROJECT_BINARY_DIR}/arpack-ng-config-version.cmake" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/arpack${SUFFIX64}-ng)
```

Comment:
There should be no need to impact version cmake file.

Reply:
From a packaging point of view, it would be nice to allow users to install either the 32-bit indexing version or the 64-bit indexing version and have a complete package. At the same time, it would be nice to allow users to install both version of ARPACK at the same time without file conflicts.
IIUC, that would be easier this file would be duplicated. 
Is there a better solution for the above use cases?

Reply:
For me, you already have everything you need as is. Install with `./configure --prefix=/path/to/arpack-ng/local` and `INTERFACE64=1 ./configure --prefix=/path/to/arpack-ng/local64`. Use with `module load arpack` or `module load arpack64` : no code impact, but just adjusted environment (`CMAKE_MODULE_PATH` adjusted by `module load`).

Reply:
I agree that this is a perfectly acceptable solution for someone that compiles ARPACK from sources themselves.
From a distribution/packaging point of view, it would be preferable if both versions (32-bit indexing and 64-bit indexing) could be installed at the default prefix.
64-bit indexing in the "classic" numeric math libraries is still used quite infrequently (at least that's my impression). But with memory getting cheaper and a more frequent usage on more "beefy" shared HW, it looks like its application becomes more widespread.
Supporting the packaging and distribution of both versions of ARPACK might also be interesting for other distributions (like e.g. Debian based distros which currently only package the 32-bit indexing version afaict).